torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -3542,10 +3542,10 @@ def _calc_grad_norm(parameters: List[torch.nn.Parameter], p: float) -> torch.Ten
         local_norm = torch.tensor(max(par.grad.detach().abs().max() for par in parameters))
     else:
         # Compute the norm in full precision no matter what
-        local_norm = torch.linalg.norm(
+        local_norm = torch.linalg.vector_norm(
             torch.stack(
                 [
-                    torch.linalg.norm(par.grad.detach(), p, dtype=torch.float32)
+                    torch.linalg.vector_norm(par.grad.detach(), p, dtype=torch.float32)
```

Comment:
@fduwjj Do you really want to use `linalg.norm` here, or is this change correct? You were sometimes dispatching to the `matrix_norm` function, and that's why you were getting the errors you were getting of `p=1.3` and `2.5`. Could you confirm that `vector_norm` is actually what you wanted to use, and that this change is indeed correct?

Also, as a side note, if the parameters here are `float64` or complex, this operation will error out because of the hardcorded `float32` will error out.

Reply:
In general, could you please review all the changes to distributed that this PR introduces? Thank you!

Reply:
Opps, just saw this, I think it looks good. thanks!